Amazon.IonObjectMapper/IonObjectSerializer.cs

```diff
@@ -367,17 +370,34 @@ private PropertyInfo FindProperty(string readName)
             return GetValidProperties(false).FirstOrDefault(p => String.Equals(p.Name, name));
         }
 
-        private bool IsReadOnlyProperty(PropertyInfo property)
+        private static bool IsReadOnlyProperty(PropertyInfo property)
         {
             return property.SetMethod == null;
         }
         
         private FieldInfo FindField(string name)
         {
             var exact = targetType.GetField(name, BINDINGS);
-            if (exact != null && IsField(exact))
+            if (exact != null)
```

Comment:
this can be simplified to:
```
if (exact != null && IsField(exact)) {
   return exact;
}
...
```

Reply:
Nope because the following else condition is an else for `exact != null`, not for `exact != null && IsField(exact)`.